src/web/app/pages-session/session-submission-page/session-submission-page.component.ts

```diff
@@ -955,4 +997,120 @@ export class SessionSubmissionPageComponent implements OnInit, AfterViewInit {
+            questionId: model.feedbackQuestionId,
+            intent: this.intent,
+            key: this.regKey,
+            moderatedPerson: this.moderatedPerson,
+            previewAs: this.previewAsPerson,
+          }));
+        }
+      });
+
+      // Find the groupable and ungroupable questions and construct the recipient to question mapping.
+      forkJoin(recipientsObservables)
+          .pipe(finalize(() => {
+            this.ungroupableQuestionsSorted = Array.from(this.ungroupableQuestions).sort();
+            this.hasLoadedAllRecipients = true;
+          }))
+          .subscribe({
+            next: (recipients: FeedbackQuestionRecipients[]) => {
+                for (let i = 0; i < recipients.length; i += 1) {
+                  const question: QuestionSubmissionFormModel = questionsToBeLoaded[i];
+                  // Only questions with question submission form mode being FIXED_RECIPIENT and with question type
+                  // not being CONSTSUM_RECIPIENTS, RANK_RECIPIENTS, and CONTRIB, are the groupable questions.
+                  if (this.getQuestionSubmissionFormMode(question, recipients[i].recipients.length)
+                      === QuestionSubmissionFormMode.FIXED_RECIPIENT
+                      && question.questionType !== FeedbackQuestionType.CONSTSUM_RECIPIENTS
+                      && question.questionType !== FeedbackQuestionType.RANK_RECIPIENTS
+                      && question.questionType !== FeedbackQuestionType.CONTRIB) {
+
+                    for (let j = 0; j < recipients[i].recipients.length; j += 1) {
+                      const recipient: FeedbackQuestionRecipient = recipients[i].recipients[j];
+                      this.addQuestionForRecipient(recipient.identifier, question.questionNumber);
+                    }
+                  } else {
+                    this.ungroupableQuestions.add(question.questionNumber);
+                  }
+                }
+              },
+              error: () => {
+                this.statusMessageService.showWarningToast('Failed to build groupable questions');
+              },
+            },
+          );
+    }
+  }
+
+  /**
+   * Gets recipient name in {@code FIXED_RECIPIENT} mode and in {@code GROUP_RECIPIENTS} view.
+   */
+  getRecipientName(recipientIdentifier: string): string {
+    const question: QuestionSubmissionFormModel | undefined =
+        this.questionSubmissionForms.find((model: QuestionSubmissionFormModel) =>
+            model.questionNumber === this.recipientQuestionMap.get(recipientIdentifier)!.values().next().value);
+
+    if (!question) {
+      this.statusMessageService.showErrorToast('Failed to build groupable questions');
+      return 'Unknown';
+    }
+
+    const recipient: FeedbackResponseRecipient | undefined =
+        question!.recipientList.find(
+            (r: FeedbackResponseRecipient) => r.recipientIdentifier === recipientIdentifier);
+
+    return recipient ? recipient.recipientName : 'Unknown';
```

Comment:
Is Unknown an expected case? Seems like unknown would mean a bug in our application instead. If that's the case, we shouldn't silently replace with unknown. Users likely won't be able to answer the question if the recipient is not known.

Reply:
This method is mirroring the behaviour of `getRecipientName` in `question-submission-form.component.ts` (line 287). This follows the original implementation and I am also not sure whether there will be a legitimate case where the recipient will be "unknown".